asv/plugins/virtualenv.py

```diff
@@ -124,13 +124,17 @@ def install_requirements(self):
 
         self.setup()
 
-        self.upgrade('setuptools==3.8')
-
-        for key, val in six.iteritems(self._requirements):
-            if val is not None:
-                self.upgrade("{0}=={1}".format(key, val))
-            else:
-                self.upgrade(key)
+        self._run_executable('pip', ['install', '--upgrade',
+                                     'setuptools==3.8'])
```

Comment:
Yeah -- I think that's probably best.  `setuptools` can end up changing the behavior of `pip` itself, so it probably does make sense to upgrade that first, then move along with the project-specific stuff.